src/java.base/share/classes/javax/crypto/spec/HPKEParameterSpec.java

```diff
@@ -0,0 +1,436 @@
+ * so that the recipient can recover this shared secret. On the sender side,
+ * after the cipher is initialized, the key encapsulation message can be
+ * retrieved directly using the {@link Cipher#getIV()} method. On the recipient
+ * side, the key encapsulation message must be provided using the
+ * {@link #encapsulation(byte[])} method.
+ * </ul>
+ * For successful interoperability, both sides need to have identical algorithm
+ * identifiers, and supply identical
+ * {@code info}, {@code psk}, and {@code psk_id} or matching authentication
+ * keys if provided. For details about HPKE modes, refer to
+ * <a href="https://www.rfc-editor.org/rfc/rfc9180.html#section-5">Section 5</a>
+ * of RFC 9180.
+ * <p>
+ * If an HPKE cipher is initialized without parameters, an
+ * {@code InvalidKeyException} will be thrown.
+ * <p>
+ * At HPKE cipher initialization, if no HPKE implementation supports the provided
+ * key type, an {@code InvalidKeyException} should be thrown. If the provided
+ * {@code HPKEParameterSpec} is not supported by any HPKE implementation,
+ * an {@code InvalidAlgorithmParameterException} will be thrown. For example:
+ * <ul>
+ * <li> An algorithm identifier is unsupported or does not match the provided key type.
+ * <li> The key encapsulation message is not provided on the receiver side.
+ * <li> An attempt to use {@code authKey(key)} is made with an incompatible key.
+ * <li> An attempt to use {@code authKey(key)} is made but {@code mode_auth}
+ *      or {@code mode_auth_psk}) is not supported by the KEM used.
+ * </ul>
+ * After initialization, both the sender and receiver can process multiple
+ * messages in sequence with repeated {@code doFinal} calls, optionally preceded
+ * by one or more {@code updateAAD} and {@code update}. Each {@code doFinal}
+ * call performs a complete HPKE encryption or decryption operation using
+ * a distinct nonce derived from an internal sequence counter, as specified by
+ * <a href="https://www.rfc-editor.org/rfc/rfc9180.html#section-5.2">Section 5.2</a>
+ * of RFC 9180.
+ * <p>
+ * As with any AEAD cipher, each {@code doFinal} call on the server side must
+ * correspond to exactly one complete ciphertext, and the number and order of
+ * calls must match on both sides. Unlike raw AEAD usage, however, an HPKE
+ * cipher manages nonce generation internally, and there is no need for the
+ * application to reinitialize the cipher with a new IV for each message.
+ * This simplifies usage while ensuring nonce uniqueness and preserving AEAD
+ * security guarantees.
+ * <p>
+ * This example shows a sender and a receiver using HPKE to securely exchange
+ * messages with an X25519 key pair.
+ * {@snippet lang=java class="PackageSnippets" region="hpke-spec-example"}
+ *
+ * @implNote This class has defined constants for some of the standard algorithm
+ * identifiers. For example, {@link #KEM_DHKEM_P_256_HKDF_SHA256},
+ * {@link #KDF_HKDF_SHA256}, and {@link #AEAD_AES_128_GCM}. An implementation
```

Comment:
Suggest combining 2 sentences and starting with "An HPKE Cipher implementation may support ..."